qutebrowser/mainwindow/statusbar/settings.py

```diff
@@ -0,0 +1,130 @@
+        super().__init__(*args, **kwargs)
+        self.parse_config()
+
+    def _test_feature(self, setting_name, tab=None, default=False):
+        if not tab:
+            tab = self._current_tab()
+        if not tab:
+            return default
+        try:
+            return tab.test_feature(setting_name)
+        except KeyError:
+            return default
+
+    def _to_bool(self, setting_name, url):
+        """Return a bool for Bool and BoolAsk settings."""
+        try:
+            opt = config.instance.get_opt(setting_name)
+        except config.configexc.Error as err:
+            raise ValueError(str(err))
+        if not opt.supports_pattern:
+            url = None
+        obj = config.instance.get_obj(setting_name, url=url)
+        if isinstance(opt.typ, configtypes.BoolAsk):
+            default = False if obj == 'ask' else obj
+            return self._test_feature(setting_name, default=default)
+        elif isinstance(opt.typ, configtypes.Bool):
+            return obj
+        raise ValueError(
+            "Setting '{}' is not a boolean setting."
+            .format(setting_name)
+        )
+
+    def parse_config(self):
+        """Parse settings from the config option.
+
+        A ValueError will be raised for any errors in parsing or
+        applying the settings.
+        """
+        raw = config.instance.get(self.config_option)
+        config_ = [x.split('|') for x in raw if x.strip()]
+        # Use `_to_bool()` to do some early validation
+        for part in config_:
+            try:
+                _indicater, value = part
+                self._to_bool(value, None)
+            except ValueError as err:
+                message.error(
+                    "Error parsing '{}' setting value {}: {}"
+                    .format(self.config_option, part, err)
+                )
+                return
+        self._config = config_
+        tab = self._current_tab()
+        if tab:
+            self.on_url_changed(tab.url())
+
+    def on_url_changed(self, url):
+        """Update the widget to reflect settings for url."""
+        if not url.isValid():
+            url = None
+        parts = [
+            t[0] for t in self._config
+            if self._to_bool(t[1], url)
+        ]
+        self.setText("[{}]".format(''.join(parts)))
+
+    def on_tab_changed(self, tab):
+        """Update tab settings text when tab changed."""
+        if not tab:
+            # Probably some timing issue with tabs/windows closing down.
+            return
+        self.on_url_changed(tab.url())
+
+    def on_load_status_changed(self, _status):
+        """Update widget when load status changed."""
+        self.on_tab_changed(self._current_tab())
+
+    def _current_tab(self):
+        """Get the currently displayed tab."""
+        # pylint: disable=protected-access
+        window = objreg.get('tabbed-browser', scope='window',
+                            window=self.parent()._win_id)
```

Comment:
It's probably cleaner to pass the `win_id` to `BooleanSettings` when creating it instead.